[2025-01-06T14:53:16.734+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: nba_data_pipeline.spark_kafka_to_postgres_players manual__2025-01-06T14:53:09.404188+00:00 [queued]>
[2025-01-06T14:53:16.743+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: nba_data_pipeline.spark_kafka_to_postgres_players manual__2025-01-06T14:53:09.404188+00:00 [queued]>
[2025-01-06T14:53:16.745+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 4
[2025-01-06T14:53:16.754+0000] {taskinstance.py:1327} INFO - Executing <Task(SparkSubmitOperator): spark_kafka_to_postgres_players> on 2025-01-06 14:53:09.404188+00:00
[2025-01-06T14:53:16.757+0000] {standard_task_runner.py:57} INFO - Started process 85 to run task
[2025-01-06T14:53:16.759+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'nba_data_pipeline', 'spark_kafka_to_postgres_players', 'manual__2025-01-06T14:53:09.404188+00:00', '--job-id', '272', '--raw', '--subdir', 'DAGS_FOLDER/full_nba_data_pipeline copy.py', '--cfg-path', '/tmp/tmphmd1tf8g']
[2025-01-06T14:53:16.763+0000] {standard_task_runner.py:85} INFO - Job 272: Subtask spark_kafka_to_postgres_players
[2025-01-06T14:53:16.948+0000] {task_command.py:410} INFO - Running <TaskInstance: nba_data_pipeline.spark_kafka_to_postgres_players manual__2025-01-06T14:53:09.404188+00:00 [running]> on host d37d09639978
[2025-01-06T14:53:16.999+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='your_email@example.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='nba_data_pipeline' AIRFLOW_CTX_TASK_ID='spark_kafka_to_postgres_players' AIRFLOW_CTX_EXECUTION_DATE='2025-01-06T14:53:09.404188+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-06T14:53:09.404188+00:00'
[2025-01-06T14:53:17.005+0000] {base.py:73} INFO - Using connection ID 'spark_default' for task execution.
[2025-01-06T14:53:17.006+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.5.4,org.postgresql:postgresql:42.6.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgresPlayers --queue root.default --deploy-mode client /usr/local/***/include/scripts/scripts/kafka_to_postgres_players.py
[2025-01-06T14:53:17.019+0000] {spark_submit.py:490} INFO - /opt/spark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-01-06T14:53:18.641+0000] {spark_submit.py:490} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-01-06T14:53:18.726+0000] {spark_submit.py:490} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2025-01-06T14:53:18.727+0000] {spark_submit.py:490} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2025-01-06T14:53:18.731+0000] {spark_submit.py:490} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12 added as a dependency
[2025-01-06T14:53:18.732+0000] {spark_submit.py:490} INFO - org.postgresql#postgresql added as a dependency
[2025-01-06T14:53:18.734+0000] {spark_submit.py:490} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-d90ca9e1-2e82-494b-9768-d43611a6f285;1.0
[2025-01-06T14:53:18.735+0000] {spark_submit.py:490} INFO - confs: [default]
[2025-01-06T14:53:21.618+0000] {spark_submit.py:490} INFO - found org.apache.spark#spark-sql-kafka-0-10_2.12;3.5.4 in central
[2025-01-06T14:53:22.261+0000] {spark_submit.py:490} INFO - found org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.5.4 in central
[2025-01-06T14:53:22.549+0000] {spark_submit.py:490} INFO - found org.apache.kafka#kafka-clients;3.4.1 in central
[2025-01-06T14:53:22.732+0000] {spark_submit.py:490} INFO - found org.lz4#lz4-java;1.8.0 in central
[2025-01-06T14:53:22.988+0000] {spark_submit.py:490} INFO - found org.xerial.snappy#snappy-java;1.1.10.5 in central
[2025-01-06T14:53:24.141+0000] {spark_submit.py:490} INFO - found org.slf4j#slf4j-api;2.0.7 in central
[2025-01-06T14:53:25.648+0000] {spark_submit.py:490} INFO - found org.apache.hadoop#hadoop-client-runtime;3.3.4 in central
[2025-01-06T14:53:25.863+0000] {spark_submit.py:490} INFO - found org.apache.hadoop#hadoop-client-api;3.3.4 in central
[2025-01-06T14:53:27.432+0000] {spark_submit.py:490} INFO - found commons-logging#commons-logging;1.1.3 in central
[2025-01-06T14:53:27.703+0000] {spark_submit.py:490} INFO - found com.google.code.findbugs#jsr305;3.0.0 in central
[2025-01-06T14:53:29.736+0000] {spark_submit.py:490} INFO - found org.apache.commons#commons-pool2;2.11.1 in central
[2025-01-06T14:53:29.762+0000] {spark_submit.py:490} INFO - found org.postgresql#postgresql;42.6.0 in central
[2025-01-06T14:53:29.770+0000] {spark_submit.py:490} INFO - found org.checkerframework#checker-qual;3.31.0 in central
[2025-01-06T14:53:29.866+0000] {spark_submit.py:490} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-sql-kafka-0-10_2.12/3.5.4/spark-sql-kafka-0-10_2.12-3.5.4.jar ...
[2025-01-06T14:53:30.137+0000] {spark_submit.py:490} INFO - [SUCCESSFUL ] org.apache.spark#spark-sql-kafka-0-10_2.12;3.5.4!spark-sql-kafka-0-10_2.12.jar (359ms)
[2025-01-06T14:53:30.312+0000] {spark_submit.py:490} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-token-provider-kafka-0-10_2.12/3.5.4/spark-token-provider-kafka-0-10_2.12-3.5.4.jar ...
[2025-01-06T14:53:30.418+0000] {spark_submit.py:490} INFO - [SUCCESSFUL ] org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.5.4!spark-token-provider-kafka-0-10_2.12.jar (279ms)
[2025-01-06T14:53:30.504+0000] {spark_submit.py:490} INFO - downloading https://repo1.maven.org/maven2/org/apache/kafka/kafka-clients/3.4.1/kafka-clients-3.4.1.jar ...
[2025-01-06T14:53:32.233+0000] {spark_submit.py:490} INFO - [SUCCESSFUL ] org.apache.kafka#kafka-clients;3.4.1!kafka-clients.jar (1815ms)
[2025-01-06T14:53:32.330+0000] {spark_submit.py:490} INFO - downloading https://repo1.maven.org/maven2/com/google/code/findbugs/jsr305/3.0.0/jsr305-3.0.0.jar ...
[2025-01-06T14:53:32.419+0000] {spark_submit.py:490} INFO - [SUCCESSFUL ] com.google.code.findbugs#jsr305;3.0.0!jsr305.jar (185ms)
[2025-01-06T14:53:32.503+0000] {spark_submit.py:490} INFO - downloading https://repo1.maven.org/maven2/org/apache/commons/commons-pool2/2.11.1/commons-pool2-2.11.1.jar ...
[2025-01-06T14:53:32.612+0000] {spark_submit.py:490} INFO - [SUCCESSFUL ] org.apache.commons#commons-pool2;2.11.1!commons-pool2.jar (193ms)
[2025-01-06T14:53:32.697+0000] {spark_submit.py:490} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-runtime/3.3.4/hadoop-client-runtime-3.3.4.jar ...
[2025-01-06T14:53:44.967+0000] {spark_submit.py:490} INFO - :: resolution report :: resolve 11043ms :: artifacts dl 15188ms
[2025-01-06T14:53:44.973+0000] {spark_submit.py:490} INFO - :: modules in use:
[2025-01-06T14:53:44.974+0000] {spark_submit.py:490} INFO - com.google.code.findbugs#jsr305;3.0.0 from central in [default]
[2025-01-06T14:53:44.976+0000] {spark_submit.py:490} INFO - commons-logging#commons-logging;1.1.3 from central in [default]
[2025-01-06T14:53:44.977+0000] {spark_submit.py:490} INFO - org.apache.commons#commons-pool2;2.11.1 from central in [default]
[2025-01-06T14:53:44.981+0000] {spark_submit.py:490} INFO - org.apache.hadoop#hadoop-client-api;3.3.4 from central in [default]
[2025-01-06T14:53:44.981+0000] {spark_submit.py:490} INFO - org.apache.hadoop#hadoop-client-runtime;3.3.4 from central in [default]
[2025-01-06T14:53:44.983+0000] {spark_submit.py:490} INFO - org.apache.kafka#kafka-clients;3.4.1 from central in [default]
[2025-01-06T14:53:44.983+0000] {spark_submit.py:490} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12;3.5.4 from central in [default]
[2025-01-06T14:53:44.984+0000] {spark_submit.py:490} INFO - org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.5.4 from central in [default]
[2025-01-06T14:53:44.985+0000] {spark_submit.py:490} INFO - org.checkerframework#checker-qual;3.31.0 from central in [default]
[2025-01-06T14:53:44.986+0000] {spark_submit.py:490} INFO - org.lz4#lz4-java;1.8.0 from central in [default]
[2025-01-06T14:53:44.989+0000] {spark_submit.py:490} INFO - org.postgresql#postgresql;42.6.0 from central in [default]
[2025-01-06T14:53:44.990+0000] {spark_submit.py:490} INFO - org.slf4j#slf4j-api;2.0.7 from central in [default]
[2025-01-06T14:53:44.991+0000] {spark_submit.py:490} INFO - org.xerial.snappy#snappy-java;1.1.10.5 from central in [default]
[2025-01-06T14:53:44.992+0000] {spark_submit.py:490} INFO - ---------------------------------------------------------------------
[2025-01-06T14:53:44.992+0000] {spark_submit.py:490} INFO - |                  |            modules            ||   artifacts   |
[2025-01-06T14:53:44.992+0000] {spark_submit.py:490} INFO - |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-01-06T14:53:44.994+0000] {spark_submit.py:490} INFO - ---------------------------------------------------------------------
[2025-01-06T14:53:44.994+0000] {spark_submit.py:490} INFO - |      default     |   13  |   11  |   11  |   0   ||   13  |   5   |
[2025-01-06T14:53:44.999+0000] {spark_submit.py:490} INFO - ---------------------------------------------------------------------
[2025-01-06T14:53:44.999+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:44.999+0000] {spark_submit.py:490} INFO - :: problems summary ::
[2025-01-06T14:53:44.999+0000] {spark_submit.py:490} INFO - :::: WARNINGS
[2025-01-06T14:53:45.000+0000] {spark_submit.py:490} INFO - [FAILED     ] org.apache.hadoop#hadoop-client-runtime;3.3.4!hadoop-client-runtime.jar: Downloaded file size (0) doesn't match expected Content Length (30085504) for https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-runtime/3.3.4/hadoop-client-runtime-3.3.4.jar. Please retry. (12323ms)
[2025-01-06T14:53:45.001+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.002+0000] {spark_submit.py:490} INFO - [FAILED     ] org.apache.hadoop#hadoop-client-runtime;3.3.4!hadoop-client-runtime.jar: Downloaded file size (0) doesn't match expected Content Length (30085504) for https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-runtime/3.3.4/hadoop-client-runtime-3.3.4.jar. Please retry. (12323ms)
[2025-01-06T14:53:45.005+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.005+0000] {spark_submit.py:490} INFO - ==== central: tried
[2025-01-06T14:53:45.009+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.010+0000] {spark_submit.py:490} INFO - https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-runtime/3.3.4/hadoop-client-runtime-3.3.4.jar
[2025-01-06T14:53:45.010+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.010+0000] {spark_submit.py:490} INFO - ::::::::::::::::::::::::::::::::::::::::::::::
[2025-01-06T14:53:45.010+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.011+0000] {spark_submit.py:490} INFO - ::              FAILED DOWNLOADS            ::
[2025-01-06T14:53:45.011+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.011+0000] {spark_submit.py:490} INFO - :: ^ see resolution messages for details  ^ ::
[2025-01-06T14:53:45.011+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.012+0000] {spark_submit.py:490} INFO - ::::::::::::::::::::::::::::::::::::::::::::::
[2025-01-06T14:53:45.012+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.014+0000] {spark_submit.py:490} INFO - :: org.apache.hadoop#hadoop-client-runtime;3.3.4!hadoop-client-runtime.jar
[2025-01-06T14:53:45.015+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.015+0000] {spark_submit.py:490} INFO - ::::::::::::::::::::::::::::::::::::::::::::::
[2025-01-06T14:53:45.015+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.016+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.028+0000] {spark_submit.py:490} INFO - 
[2025-01-06T14:53:45.031+0000] {spark_submit.py:490} INFO - :: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
[2025-01-06T14:53:45.031+0000] {spark_submit.py:490} INFO - Exception in thread "main" java.lang.RuntimeException: [download failed: org.apache.hadoop#hadoop-client-runtime;3.3.4!hadoop-client-runtime.jar]
[2025-01-06T14:53:45.032+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmitUtils$.resolveMavenCoordinates(SparkSubmit.scala:1613)
[2025-01-06T14:53:45.033+0000] {spark_submit.py:490} INFO - at org.apache.spark.util.DependencyUtils$.resolveMavenDependencies(DependencyUtils.scala:185)
[2025-01-06T14:53:45.034+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.prepareSubmitEnvironment(SparkSubmit.scala:339)
[2025-01-06T14:53:45.038+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:969)
[2025-01-06T14:53:45.039+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:199)
[2025-01-06T14:53:45.039+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:222)
[2025-01-06T14:53:45.041+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:91)
[2025-01-06T14:53:45.041+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1125)
[2025-01-06T14:53:45.042+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1134)
[2025-01-06T14:53:45.042+0000] {spark_submit.py:490} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2025-01-06T14:53:45.104+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.5.4,org.postgresql:postgresql:42.6.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgresPlayers --queue root.default --deploy-mode client /usr/local/***/include/scripts/scripts/kafka_to_postgres_players.py. Error code is: 1.
[2025-01-06T14:53:45.123+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=nba_data_pipeline, task_id=spark_kafka_to_postgres_players, execution_date=20250106T145309, start_date=20250106T145316, end_date=20250106T145345
[2025-01-06T14:53:45.169+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 272 for task spark_kafka_to_postgres_players (Cannot execute: spark-submit --master spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.5.4,org.postgresql:postgresql:42.6.0 --executor-memory 2G --driver-memory 1G --name SparkKafkaToPostgresPlayers --queue root.default --deploy-mode client /usr/local/***/include/scripts/scripts/kafka_to_postgres_players.py. Error code is: 1.; 85)
[2025-01-06T14:53:45.232+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2025-01-06T14:53:45.312+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
